### PR TITLE
Implemented encryption for the REST API Key

### PR DESCRIPTION
**Pull request by bertold**
Updated messages referencing journey to test
Miscellaneous code cleanup

### CODE REVIEW

File: src/main/java/com/mabl/integration/jenkins/JenkinsModule.java [hunk 1]
```diff
@@ -28,9 +33,13 @@ protected void configure() {
             try {
                 inputStream.close();
             } catch (IOException e) {
-                throw new RuntimeException(e);
+                exception = new RuntimeException(e);
```
[the remark this answers is not in the document]
Comment (bertold) [reply src/main/java/com/mabl/integration/jenkins/JenkinsModule.java]: It is a bad practice to throw an exception from a finally block.

File: src/main/java/com/mabl/integration/jenkins/MablStepBuilder.java [hunk 2]
```diff
@@ -67,7 +68,7 @@ public MablStepBuilder(
             final String applicationId,
             final Set<String> labels
     ) {
-        this.restApiKey = trimToNull(restApiKey);
+        this.restApiKey = Secret.fromString(trimToNull(restApiKey));
```
[the remark this answers is not in the document]
Comment (bertold) [reply src/main/java/com/mabl/integration/jenkins/MablStepBuilder.java]: I just tested this. The plugin can read the configuration and next time the job is saved, the encrypted secret is saved. This is how I tested it:
- installed Jenkins and the current version of the plugin
- created a job with a mabl step and saved the configuration
- verified that the clear-text API key is saved
- uninstalled the mabl plugin and restarted Jenkins
- verified that the job's config.xml still shows the clear-text API key
- installed my local version of the plugin
- opened the job configuration page and verified that the key still showed up
- when I hit apply, then the configuration was saved using an encrypted secret

File: src/main/java/com/mabl/integration/jenkins/MablStepBuilder.java [hunk 3]
```diff
@@ -155,16 +156,16 @@ public void perform(
...
             e.printStackTrace(outputStream);
             if(continueOnMablError) {
                 run.setResult(Result.FAILURE);
             } else {
                 run.setResult(Result.SUCCESS);
             }
         } catch (TimeoutException e) {
-            outputStream.printf("Oh dear. Your journeys exceeded the max plugin runtime limit of %d seconds.%n" +
-                    "We've aborted this Jenkins step, but your journeys may still be running in mabl.", EXECUTION_TIMEOUT_SECONDS);
+            outputStream.printf("Oh dear. Your tests exceeded the max plugin runtime limit of %d seconds.%n" +
```
Comment (DonMcNamara) [inline src/main/java/com/mabl/integration/jenkins/MablStepBuilder.java]: Not really part of your PR, but I'm not sure I like the conversational tone of "Oh dear." here, but that might just be me.

File: README.md [hunk 4]
```diff
@@ -139,6 +139,11 @@ Send build environment variables is set. Collecting the following information:
 
 ### Change Log
 
+#### v0.0.17 (29 January 2020)
+-   API key is now encrypted in configuration file
```
Comment (DonMcNamara) [inline README.md]: Should this be a minor version bump?

File: src/main/java/com/mabl/integration/jenkins/JenkinsModule.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (DonMcNamara) [reply src/main/java/com/mabl/integration/jenkins/JenkinsModule.java]: What java versions do we support? Can this be replaced with a try with resources? https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

File: src/main/java/com/mabl/integration/jenkins/MablStepBuilder.java (hunk 3 above)
Comment (twistedpair) [reply src/main/java/com/mabl/integration/jenkins/MablStepBuilder.java]: Probably good to temper the voice here. We're no longer trying to be a javial, personified machine TTBOMK.

File: src/main/java/com/mabl/integration/jenkins/JenkinsModule.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (twistedpair) [reply src/main/java/com/mabl/integration/jenkins/JenkinsModule.java]: I would agree if this was a `finally` block or finalizer, but it isn't. Compared to the added complexity of trying to effect the same behavior passing around, checking for, and rethrowing exceptions, the common pattern seems simpler and more robust. Just dug into Josha Boch again this is an accepted pattern for exception handling.

File: src/main/java/com/mabl/integration/jenkins/MablStepBuilder.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (twistedpair) [reply src/main/java/com/mabl/integration/jenkins/MablStepBuilder.java]: Thanks. Is it possible to test without uninstalling the plugin?

[the remark this answers is not in the document]
Comment (bertold) [reply src/main/java/com/mabl/integration/jenkins/MablStepBuilder.java]: Even though I uninstalled the plugin, the job itself did not discard the old data (i.e. I could see the API key in the job configuration file). I did not see an obvious way how to upgrade to a dev version, to be honest.

File: src/main/java/com/mabl/integration/jenkins/MablStepBuilder.java (hunk 3 above)
Comment (bertold) [reply src/main/java/com/mabl/integration/jenkins/MablStepBuilder.java]: Let me work on this on a follow-up PR.

File: README.md (hunk 4 above)
Comment (bertold) [reply README.md]: I feel this is a minor change.

File: src/main/java/com/mabl/integration/jenkins/JenkinsModule.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (bertold) [reply src/main/java/com/mabl/integration/jenkins/JenkinsModule.java]: Frankly, I don't care one way or the other. At least, there is one plugin (checkstyle) that flags this as an issue. I still believe it is a bad pattern. I will revert it. @DonMcNamara - we are expected to use Java 6 language level for this version of the plugin.

[the remark this answers is not in the document]
Comment (twistedpair) [reply src/main/java/com/mabl/integration/jenkins/JenkinsModule.java]: I'd done this review without using an IDE and all code deltas shown by GH didn't reveal this was in a nested `try/finally` block. My apologies for the confusion.

**Pull request merged**